main.ts

```diff
@@ -158,81 +175,23 @@ async function handleConnection(conn: Deno.Conn): Promise<void> {
 
         // Write everything else from the connection
         const contentLength = parseInt(headers.get("content-length")!);
-        while (bodyBytesRead < contentLength) { 
+        while (bodyBytesRead < contentLength) {
             const nbytes = await conn.read(body);
             if (nbytes === null) {
                 break;
             }
             (await backendConn).write(body.slice(0, nbytes));
             bodyBytesRead += nbytes;
         }
-      
-        //log successful request
-        const referer : string | undefined = headers.get("referer");
-        const userAgent : string | undefined = headers.get("user-agent");
 
-        const accessLog : AccessLog = {
-            method: requestLine.split(" ")[0],
-            url: requestLine.split(" ")[1],
-            protocol: requestLine.split(" ")[2],
-            status: 200,
-            size: contentLength,
-            referer: referer? referer : "",
-            userAgent: userAgent? userAgent : "",
-            responseTime: 0,
-            upstream_response_time: 0,
-            backend: destIp[0].address
+        const serverReadableStream = backendConn.readable;
+        const response = new Response(serverReadableStream);
+        // Add cache using a clone of the response; don't await this as it takes a while
+        addCache(address, response.clone());
+        if (response) {
+            await response.body?.pipeTo(conn.writable);
+            break sendConnection;
         }
-
```

Comment:
Make sure to rebase your branch onto main - this code should be in your final PR